pkg/providers/aws/cluster_network_provider.go

```diff
@@ -1510,18 +1523,90 @@ func getNetworkProviderConfig(ctx context.Context, configManager ConfigManager,
+			return nil, errorUtil.Wrap(err, "failed to parse cidr block from _network strategy")
+		}
+		logger.Infof("found vpc cidr block %s in network strategy tier %s", vpcCidr.String(), tier)
+		return vpcCidr, nil
 	}
 
-	_, vpcCidr, err := net.ParseCIDR(*vpcCreateConfig.CidrBlock)
+	//if vpcCreateConfig.CidrBlock is nil or an empty string we can go ahead and set a sensible non overlapping default with a mask size of /26
+	defaultCIDR, err := n.getNonOverlappingDefaultCIDR(ctx)
 	if err != nil {
-		return nil, errorUtil.Wrap(err, "failed to parse cidr block from _network strategy")
+		return nil, errorUtil.Wrap(err, "failed to generate default CIDR")
 	}
-	cidrMask, _ := vpcCidr.Mask.Size()
 
-	logger.Infof("found vpc cidr block %s/%d in network strategy tier %s", vpcCidr.IP.String(), cidrMask, tier)
-	return vpcCidr, nil
+	// a default cidr is generated and updated to the config map, that is returned for use in the network creation
+	return defaultCIDR, nil
+}
+
+// getNonOverlappingDefaultCIDR returns a non overlapping cidr block based on the OpenShift vpc cidr block
+// the default mask is /26
+// for other masks the user is required to provide their own via config
+func (n *NetworkProvider) getNonOverlappingDefaultCIDR(ctx context.Context) (*net.IPNet, error) {
+	clusterVpc, err := getClusterVpc(ctx, n.Client, n.Ec2Api, n.Logger)
+	if err != nil {
+		errMsg := "failed to get cluster vpc for cidr block"
+		return nil, errorUtil.Wrap(err, errMsg)
+	}
+	//parse the cidr to an ipnet cidr in order to manipulate the ip
+	_, clusterNet, err := net.ParseCIDR(aws.StringValue(clusterVpc.CidrBlock))
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "error parsing cluster cidr block")
+	}
+
+	// this map is used to loop through the available options for a vpc cidr range in aws
+	// See aws docs https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Subnets.html#vpc-sizing-ipv4
+	cidrRanges := map[string]string{
+		"10.255.255.255/8":  "10.0.0.0/26",
+		"172.31.255.255/12": "172.16.0.0/26",
+	}
+
+	// in this loop we loop through the available cidr ranges for vpcs in aws
+	// in each range the ip of the cidr block is incremented only in the A and B range
+	// the reason for this is that cro allows a maximum mask size of /16
+	// the default mask size set is /26
+	//
+	// the current logic checks that the cidr block does not overlap with the cluster machine cidr range
+	// there is a follow on JIRA to check overlap with pod and service cidr blocks -> https://issues.redhat.com/MDGAPI-1000
+	for cidrRange, potentialDefault := range cidrRanges {
+		_, cidrRangeNet, err := net.ParseCIDR(cidrRange)
+		if err != nil {
+			fmt.Println("error ", err)
+		}
+
+		// our potential default
+		potentialDefaultIP, _, err := net.ParseCIDR(potentialDefault)
+		if err != nil {
+			fmt.Println("error ", err)
+		}
+		// loop for as long as potential default is a valid CIDR in range
+		for cidrRangeNet.Contains(potentialDefaultIP) {
+			fmt.Println("inside the loop")
```

Comment:
I think this could go?

Reply:
🤦‍♀️ yep

Reply:
I liked it, can we add this back in, plzzzz.